### PR TITLE
Linux: check processes for using deleted shared libraries

### PR DESCRIPTION
**Pull request by cgzones**
Shared libraries can be replaced by an upgrade, highlight processes
using deleted shared libraries.

Link with highlightDeletedExe setting, enabled by default.

Currently only checked on Linux.

### CODE REVIEW

**Comment by BenBE** [comment]
What about combining this with #414?

**Review by BenBE — COMMENTED** [review]
Patch itself LGTM, but I still wonder if we really should split the two features, especially as #414 more or less implies this flag is also triggered.

**Comment by cgzones** [comment]
Postpone after #414 to merge enable-logic with deleted executables.

**Comment by alicektx** [comment]
...really really thank you for doing / implementing this ability into htop! had been using [this](https://github.com/gsauthof/utility#oldprocs) so far to check such...

**Comment by BenBE** [comment]
@cgzones Could you update this now that #414 is merged?
Please note this earlier [review comment](ttps://github.com/htop-dev/htop/pull/595#pullrequestreview-632706242).

File: linux/LinuxProcessList.c
Comment (BenBE) [inline linux/LinuxProcessList.c:605]: Mind this bugfix:
https://github.com/htop-dev/htop/commit/51ecc62d92d78d8ac5a3367e0ca8f4865880319d

You need to track the state transition to mark for `makeCommandStr`.

**Review by BenBE — CHANGES_REQUESTED** [review]

File: linux/LinuxProcessList.c
Comment (BenBE) [inline linux/LinuxProcessList.c:1355]: Only reset this flag, once we actually re-calculate it. Maybe put directly before call to `LinuxProcessList_readMaps`?

**Review by BenBE — CHANGES_REQUESTED** [review]

File: linux/LinuxProcessList.c
Comment (cgzones) [reply linux/LinuxProcessList.c]: true, thanks for catching

Comment (BenBE) [reply linux/LinuxProcessList.c]: I think that's why we do those code reviews ;-) Less stuff we need to fix later. :P

**Review by BenBE — APPROVED** [review]

**Pull request merged**
